src/Chart/Pie.php

```diff
@@ -2338,4 +2318,48 @@ public function clean0Values(array $Data, array $Palette, $DataSerie, $AbscissaS
 
         return [$Data, $NewPalette];
     }
+
+    /**
+     * Returns the value to display on a pie chart.
+     *
+     * @param int|float $Value
+     *   The value of current item.
+     * @param int $WriteValues
+     *   The type of value to write:
+     *   - PIE_VALUE_NATURAL for an absolute value;
+     *   - PIE_VALUE_PERCENTAGE for a percentage value;
+     *   - PIE_VALUE_BOTH for both an absolute and percentage value.
+     * @param int $SerieSum
+     *   The sum of all items.
+     * @param int $Precision
+     *   The number of decimal digits to round to, in case of displaying a percentage.
+     * @param string $ValueSuffix
+     *   The text to display after the value, in case an absolute value gets displayed.
+     *
+     * @return string
+     *   The value to use for display.
+     */
+    protected function getDisplayValue($Value, $WriteValues, $SerieSum, $Precision, $ValueSuffix)
+    {
+        if ($WriteValues == PIE_VALUE_NATURAL || $WriteValues == PIE_VALUE_BOTH) {
```

Comment:
Hello, sorry for late reply. Try the code below, it should silence any static code exceptions.

```php
$ValidWriteValues = [PIE_VALUE_NATURAL, PIE_VALUE_PERCENTAGE, PIE_VALUE_BOTH];
if (false === in_array($WriteValues, $ValidWriteValues, true)) {
    return '';
}

$isBothOrNatural = in_array($WriteValues, [PIE_VALUE_NATURAL, PIE_VALUE_BOTH], true);
$isBothOrPercentage = in_array($WriteValues, [PIE_VALUE_PERCENTAGE, PIE_VALUE_BOTH], true);

$Absolute = true === $isBothOrNatural ? ($Value . $ValueSuffix) : null;
$Percentage = true === $isBothOrPercentage
    ? sprintf('%s%', round((100 / $SerieSum) * $Value, $Precision))
    : null
;

if (null === $Absolute && null === $Percentage) {
    throw new RuntimeException(
        "Neither absolute nor percentage display value was calculated"
        . " for display value \"{$WriteValues}\"."
    );
}

if (null !== $Absolute && null !== $Percentage) {
    $Result = "{$Absolute}\n({$Percentage})";
} elseif (null !== $Absolute) {
    $Result = $Absolute;
} elseif (null !== $Percentage) {
    $Result = $Percentage;
}

return $Result;
```
Also, can you please write a test case in [this](https://github.com/szymach/c-pchart/blob/3.0/tests/unit/PieTest.php) unit test for the new option?